### PR TITLE
Additional comment features

### PR DESCRIPTION
**Pull request by kpancha**
Step 6 of the week 3 walkthrough

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java [hunk 1]
```diff
@@ -42,11 +42,17 @@
...
+    int maxNumComments = maxCommentParam == null ? Integer.MAX_VALUE : Integer.parseInt(maxCommentParam);
+    
     PreparedQuery results = datastore.prepare(query);
 
+    int i = 0;
     List<Comment> commentsList = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
+      if (i == maxNumComments) {
+        break;
+      }
```
Comment (jesseelizabeth) [inline portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java:55]: Let's make this a for loop where i < the min of either results.length or maxComments, then you don't have to break, it will just stop

File: portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java [hunk 2]
```diff
@@ -42,11 +44,14 @@
...
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-
+    String maxCommentParam = request.getParameter("max-comments");
+    int maxNumComments = maxCommentParam == null ? Integer.MAX_VALUE : Integer.parseInt(maxCommentParam);
+    
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> commentsList = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    for (int i = 0; i < results.countEntities(FetchOptions.Builder.withDefaults()) && i < maxNumComments; i++) {
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java:53]: Lets pull result.countEntities(..options..) into a local variable right before this loop 
Based on the javadocs for this api (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#countentities) --> it seems like it could be a somewhat expensive operation, so we shouldn't recompute at every loop iteration

File: portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java [hunk 3]
```diff
@@ -42,11 +44,14 @@
...
+      Entity entity = results.asList(FetchOptions.Builder.withDefaults()).get(i);
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java:54]: also we can pull FetchOptions.builder.withDefault into a local var as well

File: portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java [hunk 4]
```diff
@@ -56,6 +61,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Comment comment =
           new Comment(name, content, numLikes, timestamp, KeyFactory.keyToString(key));
       commentsList.add(comment);
+      i++;
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java:64]: why is this needed?

Comment (kpancha) [reply portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java:64]: Typo forgot to take it out
